crates/sui-framework/packages/sui-framework/sources/crypto/zklogin.move

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::zklogin {
+    use std::string::String;
+    use sui::object;
+    use sui::object::UID;
+    use sui::tx_context::TxContext;
+    use sui::tx_context;
+
+    /// Error if any of the inputs are longer than the allowed upper bounds.
+    const EInvalidInput: u64 = 0;
+
+    /// Error if the proof consisting of the inputs provided to the verification function is invalid.
+    const EInvalidProof: u64 = 1;
+
+    /// Posession of a VerifiedID proves that the user's address was created using zklogin and the given parameters.
+    struct VerifiedID has key {
+
+        /// The ID of this VerifiedID
+        id: UID,
+
+        /// The address this VerifiedID is associated with
+        address: address,
+
+        /// The name of the key claim
+        kc_name: String,
+
+        /// The value of the key claim
+        kc_value: String,
+
+        /// The issuer
+        iss: String,
+
+        /// The audience (wallet)
+        aud: String,
+    }
+
+    /// Returns the address associated with the given VerifiedID
+    public fun get_address_from_verified_id(verified_id: &VerifiedID): address {
```

Comment:
@tnowacki, does receiver-style syntax give us a way to improve on these names? This is a classic case of "module that declares > 1 meaningful type so names are hard".

If not @jonas-lj, I might suggest separate modules for `VerifiedId` and `VerifiedIssuer` (since all the functionality can be cleanly separated.

Reply:
The PR looks good to me other than this: I think we should improve the getter names one way or another.

Reply:
That's a good idea: I've separated the modules for `VerifiedID` and `VerifiedIssuer` into two modules in the latest commit.

Reply:
Separating the two if possible, would make things easier since the method syntax will not be widely available (outside of a "nightly"/alpha build) until early next year.  

... but with the new syntax you could always rename this function for methods: `public use fun get_address_from_verified_id as VerifiedID.address`